src/simulation.jl

```diff
@@ -64,27 +64,32 @@ struct AllocVecCall <: AbstractCall
     name
     form
     dimension
+    T
+end
+
+struct AllocVecCallError <: Exception
+    c::AllocVecCall
 end
 
 # TODO: There are likely better ways of dispatching on dimension instead of
 # storing it inside an AllocVecCall.
 Base.Expr(c::AllocVecCall) = begin
-    resolved_form = @match (c.name, c.form, c.dimension) begin
-        (_, :Form0, 2) => :V
-        (_, :Form1, 2) => :E
-        (_, :Form2, 2) => :Tri
-        (_, :DualForm0, 2) => :Tri
-        (_, :DualForm1, 2) => :E
-        (_, :DualForm2, 2) => :V
-
-        (_, :Form0, 1) => :V
-        (_, :Form1, 1) => :E
-        (_, :DualForm0, 1) => :E
-        (_, :DualForm1, 1) => :V
-        _ => return :AllocVecCall_Error
+    resolved_form = @match (c.form, c.dimension) begin
```

Comment:
PR #151 also makes this refactoring and pulls it into a function to be called elsewhere, but I will leave this be and fix the merge conflict when it is time to merge 151.